cub/test/catch2_test_device_segmented_radix_sort_keys.cu

```diff
@@ -46,11 +46,50 @@
 #include "catch2_test_helper.h"
 #include "catch2_test_launch_helper.h"
 
+// TODO replace with DeviceSegmentedRadixSort::If interface once https://github.com/NVIDIA/cccl/issues/50 is addressed
+// Temporary wrapper that allows specializing the DeviceSegmentedRadixSort algorithm for different offset types
+template <bool IS_DESCENDING, typename KeyT, typename BeginOffsetIteratorT, typename EndOffsetIteratorT, typename NumItemsT>
+CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t dispatch_segmented_radix_sort_wrapper(
+  void* d_temp_storage,
+  size_t& temp_storage_bytes,
+  const KeyT* d_keys_in,
+  KeyT* d_keys_out,
+  NumItemsT num_items,
+  NumItemsT num_segments,
+  BeginOffsetIteratorT d_begin_offsets,
+  EndOffsetIteratorT d_end_offsets,
+  int begin_bit       = 0,
+  int end_bit         = sizeof(KeyT) * 8,
+  cudaStream_t stream = 0)
+{
+  cub::DoubleBuffer<KeyT> d_keys(const_cast<KeyT*>(d_keys_in), d_keys_out);
+  cub::DoubleBuffer<cub::NullType> d_values;
+  return cub::DispatchSegmentedRadixSort<
+    IS_DESCENDING,
+    KeyT,
+    cub::NullType,
+    BeginOffsetIteratorT,
+    EndOffsetIteratorT, //
+    NumItemsT>::Dispatch(d_temp_storage,
+                         temp_storage_bytes,
+                         d_keys,
+                         d_values,
+                         num_items,
+                         num_segments,
+                         d_begin_offsets,
+                         d_end_offsets,
+                         begin_bit,
+                         end_bit,
+                         false,
```

Comment:
suggestion: It would be good to test both, with `is_overwrite_okay=false` and `is_overwrite_okay=true`. Radix sort is a multi-pass algorithm, so we need to swap input and output buffers after each pass. `is_overwrite_okay=false` will need to allocate a temporary storage buffer for double-buffering between user-provided output memory and the temporary storage, increasing the total memory requirements quite a bit. 
`is_overwrite_okay=true` will use the user-provided input and output buffers for double-buffering, alleviating the need for a temporary storage buffer. In that case, we need to check `cub::DoubleBuffer<KeyT>::Current()` to figure out where the sorted output actually ended up at.

The `num_items` parameter is actually only used in the `is_overwrite_okay=false` case for computing the size of the temporary storage buffer.